sfge/src/main/java/com/salesforce/rules/DmlUtil.java

```diff
@@ -0,0 +1,93 @@
+package com.salesforce.rules;
+
+import com.salesforce.graph.EnumUtil;
+import java.util.Optional;
+import java.util.TreeMap;
+
+public class DmlUtil {
+
+    // we don't ever want this class to be initialized
+    private DmlUtil() {}
+
+    /**
+     * An enumeration of all of the possible Database.method() methods in the Database class in
+     * Apex.
+     */
+    public enum DatabaseOperation {
+        CONVERT_LEAD("Database.convertLead", true),
+        COUNT_QUERY("Database.countQuery", true),
+        COUNT_QUERY_WITH_BINDS("Database.countQueryWithBinds", true),
+        DELETE("Database.delete", true),
+        DELETE_ASYNC("Database.deleteAsync", true),
+        DELETE_IMMEDIATE("Database.deleteImmediate", true),
+        EMPTY_RECYCLE_BIN("Database.emptyRecycleBin", true),
+        EXECUTE_BATCH("Database.executeBatch", true),
+        GET_ASYNC_DELETE_RESULT("Database.getAsyncDeleteResult", true),
+        GET_ASYNC_LOCATOR("Database.getAsyncLocator", true),
+        GET_ASYNC_SAVE_RESULT("Database.getAsyncSaveResult", true),
+        GET_DELETED("Database.getDeleted", true),
+        GET_QUERY_LOCATOR("Database.getQueryLocator", true),
+        GET_QUERY_LOCATOR_WITH_BINDS("Database.getQueryLocatorWithBinds", true),
+        GET_UPDATED("Database.getUpdated", true),
+        INSERT("Database.insert", true),
+        INSERT_ASYNC("Database.insertAsync", true),
+        INSERT_IMMEDIATE("Database.insertImmediate", true),
+        MERGE("Database.merge", true),
+        QUERY("Database.query", true),
+        QUERY_WITH_BINDS("Database.queryWithBinds", true),
+        ROLLBACK("Database.rollback", true),
+        SET_SAVEPOINT("Database.setSavepoint", true),
+        UNDELETE("Database.undelete", true),
+        UPDATE("Database.update", true),
+        UPDATE_ASYNC("Database.updateAsync", true),
+        UPDATE_IMMEDIATE("Database.updateImmediate", true),
+        UPSERT("Database.upsert", true);
+
+        // Create a map of database operation method -> DatabaseOperation for fromString method
+        private static final TreeMap<String, DatabaseOperation> FROM_DATABASE_METHOD =
```

Comment:
1. Seems like this map could be a property of the top-level class instead of this enum.
2. Where possible, we try to name maps `yByX` or `xToY`, to be clear about what the mapping conceptually represents. Is there a name that meets that convention we could use instead of `FROM_DATABASE_METHOD`?

Reply:
Moved.